### PR TITLE
Add test coverage for moved api error codes - Closes #3021 #3029

### LINKED ISSUE #3021: Add test coverage for moved api error codes
### Expected behavior
Move the tests from `chain` to `http_api`

### Actual behavior
Still in `chain`, ref point: `framework/test/mocha/unit/modules/chain/modules/signatures.js:138`

### Which version(s) does this affect? (Environment, OS, etc...)
1.6.0

### PR DESCRIPTION
**Pull request by michielmulders**
### What was the problem?
No tests created for the moved logic of `postSignature` and `postTransaction`.
In short, moved error logic from chain module to http_api module so we don't need to import API specific api_error codes into chain.

On the way doing this, we found callbacks being passed together with `invoke.channel` which is not allowed. All callbacks in `invoke.channel` functions have been modified to not pass it (using `promisify` down the chain).

### How did I fix it?
Added test coverage.
Refactor `invoke.channel` by removing callbacks as argument and callbacks in `chain.js`.
Modified tests accordingly.

### How to test it?
Run tests.

### Review checklist

* The PR resolves #3021 #3029 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: framework/src/modules/chain/chain.js [hunk 1]
```diff
@@ -217,16 +217,12 @@ module.exports = class Chain {
 					action.params[2]
 				),
 			getPeers: async action =>
-				this.scope.modules.peers.shared.getPeers(
-					action.params[0],
-					action.params[1]
-				),
+				promisify(this.scope.modules.peers.shared.getPeers)(action.params[0]),
```
Comment (lsilvs) [inline framework/src/modules/chain/chain.js]: Maybe not relevant to this PR but as you tackled it already, the functions `generateDelegateList`, `getUnProcessedTransactions`, `getUnconfirmedTransactions` and `getMultisignatureTransactions` also need to be promisify'd.

Comment (michielmulders) [reply framework/src/modules/chain/chain.js]: Ah good one, let's include this in the PR as we tackled half the problem here yet.

File: framework/src/modules/http_api/controllers/peers.js [hunk 2]
```diff
@@ -60,33 +63,34 @@ PeersController.getPeers = async function(context, next) {
...
+		return next(null, {
+			data: filteredData,
+			meta: {
+				offset: filters.offset,
+				limit: filters.limit,
+				count: peersCount,
+			},
+		});
+	} catch (err) {
+		error = new ApiError(err, apiCodes.PROCESSING_ERROR);
```
Comment (limiaspasdaniel) [inline framework/src/modules/http_api/controllers/peers.js]: Why is this a `PROCESSING_ERROR`?

File: framework/src/modules/http_api/controllers/signatures.js [hunk 3]
```diff
@@ -73,12 +69,9 @@ SignaturesController.postSignature = async function(context, next) {
...
 	} catch (err) {
 		error = new ApiError(err, apiCodes.PROCESSING_ERROR);
-		console.log('inside');
-
-		context.statusCode = error.code;
-		delete error.code;
 	}
 
+	context.statusCode = error.code;
 	return next(error);
```
Comment (limiaspasdaniel) [inline framework/src/modules/http_api/controllers/signatures.js]: why is line 79 https://github.com/LiskHQ/lisk/pull/3032/commits/f26c3ebfeecc68f9e0008027cbc5b0fe791ccd72#diff-85d54af64d1362a64fb0e78e4c17ce65L79 removed?

**Comment by limiaspasdaniel** [comment]
To be merged after #3020

File: framework/src/modules/http_api/controllers/signatures.js (hunk 3 above)
Comment (michielmulders) [reply framework/src/modules/http_api/controllers/signatures.js]: Did it to have consistent behavior with transactions controller and write better tests using the `api_codes`. No breaking change.

File: framework/src/modules/http_api/controllers/peers.js (hunk 2 above)
Comment (michielmulders) [reply framework/src/modules/http_api/controllers/peers.js]: Reset back to: next(err) as this is breaking change, thanks!

**Pull request merged**